pkg/pillar/cmd/zedagent/handleconfig.go

```diff
@@ -470,6 +470,8 @@ func readConfigResponseProtoMessage(resp *http.Response, contents []byte) (bool,
 func inhaleDeviceConfig(config *zconfig.EdgeDevConfig, getconfigCtx *getconfigContext, usingSaved bool) bool {
 	log.Debugf("Inhaling config")
 
+	parseCipherContextConfig(getconfigCtx, config)
```

Comment:
This doesn't belong here. Haven't even checked the UUID.
If it isn't sufficient to reorder things in parseconfig by moving the line up 10 lines (and I suspect it is not), you need to make zedrouter handle things in different order. Note that the select loops (.e.g, in zedrouter) do not pick things up in order. select does a random selection.

Reply:
In each config, cipher context ID changes always because initial value changes for every encryption in the controller. So, the cipher context ID in cipher data of data store config and wifi config points to a new context. If we publish network config before parsing the context config in zedagent, it always fails with context not ready error. So, I am just trying to parse the context config before publishing the datastore config and network config.
Cipher context config/status is managed by zedagent only as of now.
Ref: https://github.com/lf-edge/eve/blob/42b86f2e5de5a6c414fe0e46b76cab90838f42a8/pkg/pillar/cmd/zedagent/parseconfig.go#L62